### PR TITLE
chore(docker-compose): pass correct configuration values for running SDK demo app 

### PR DESCRIPTION
**Pull request by SanchithHegde**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

This PR includes the following changes to the Docker Compose setup:

- Adds health checks for PostgreSQL and Redis containers.
- Specifies the dependencies among services.
- Specifies the correct environment variables for the `hyperswitch-web` service so that the SDK demo app works correctly.

## Motivation and Context

More robust Docker Compose setup, and enabling the SDK demo app.

## How did you test it?

Locally, with the Docker Compose setup:

Screenshot of locally running demo app

## Checklist

- [ ] I formatted the code `cargo +nightly fmt --all`
- [ ] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: docker-compose-development.yml [hunk 1]
```diff
@@ -64,6 +76,11 @@ services:
     environment:
       - CARGO_HOME=/cargo_cache
       - CARGO_TARGET_DIR=/cargo_build_cache
+    depends_on:
+      pg:
+        condition: service_healthy
+      redis-standalone:
+        condition: service_healthy
```
Comment (lsampras) [inline docker-compose-development.yml:83]: do we need healthy redis for this?

File: docker-compose.yml [hunk 2]
```diff
@@ -158,7 +179,10 @@ services:
     volumes:
       - ./config/dashboard.toml:/tmp/dashboard-config.toml
     depends_on:
-      - hyperswitch-web
+      hyperswitch-server:
```
Comment (lsampras) [inline docker-compose.yml:182]: I had left out hyperswitch-server as a dependency so people can run the control center with their local `cargo run`

File: docker-compose.yml [hunk 3]
```diff
@@ -55,14 +67,19 @@ services:
     volumes:
       - ./config:/local/config
       - ./files:/local/bin/files
+    depends_on:
+      pg:
+        condition: service_healthy
+      redis-standalone:
```
Comment (lsampras) [inline docker-compose.yml:73]: do we need healthy redis for this?

Comment (SanchithHegde) [reply docker-compose.yml:73]: Without a healthy Redis service, the application fails to start as it tries to connect to Redis at startup.

**Review by lsampras — APPROVED** [review]

**Review by Narayanbhat166 — APPROVED** [review]

**Pull request merged**